translate <optimized for change> to chinese (#304)
gaearon/overreacted.io: src/pages/optimized-for-change/index.zh-hans.md

---
title: 应变而优
date: '2018-12-12'

> spoiler: 如何写好一个API？

Reviewer: ```suggestion
spoiler: 如何写好一个API？
```
```suggestion
spoiler: 如何写好一个API？
```
```suggestion
spoiler: 如何写好一个 API？
```
Adding a space. Good translation of the title, FYI.

---

> 如何写好一个API？

Reviewer: ```suggestion
如何写好一个 API？
```

> *优秀* 的 API即好记住又明确，促进代码可读性、准确性和高效性，帮助开发者走向[成功](https://blog.codinghorror.com/falling-into-the-pit-of-success/)。

Reviewer: ```suggestion
*优秀*的 API 即好记住又明确，促进代码可读性、准确性和高效性，帮助开发者走向[成功](https://blog.codinghorror.com/falling-into-the-pit-of-success/)。
```
Add whitespaces between English and Chinese. See the guideline [here](https://github.com/mzlogin/chinese-copywriting-guidelines#%E4%B8%AD%E8%8B%B1%E6%96%87%E4%B9%8B%E9%97%B4%E9%9C%80%E8%A6%81%E5%A2%9E%E5%8A%A0%E7%A9%BA%E6%A0%BC).

> 我称这些设计方向为“第一阶层”，因为它们是类库开发者需要关注的第一件事情。你可能不得不舍弃其中一部分并作出权衡，但至少它们在你脑海中游走过。

Reviewer: ```suggestion
我称这些设计方向为「第一阶层」，因为它们是类库开发者需要关注的第一件事情。你可能不得不舍弃其中一部分并作出权衡，但至少它们在你脑海中游走过。
```
I suggest using square brackets to make it more readable. See references [here](https://github.com/mzlogin/chinese-copywriting-guidelines#%E5%85%A8%E8%A7%92%E5%92%8C%E5%8D%8A%E8%A7%92).

> 但是，只要你还未将此“流浪者”发送至火星，你的代码一直都有可能发生变化，API使用者的代码也是如此。

Reviewer: ```suggestion
但是，只要你还未将此「流浪者」发送至火星，你的代码一直都有可能发生变化，API使用者的代码也是如此。
```

> 我认识的一些优秀API开发者不会停留在像可读性这样的“第一阶层”方向上，他们将尽可能多的努力奉献给我称为“第二阶层”的API设计：**使用此API的代码如何随着时间的推移而发展**。

Reviewer: ```suggestion
我认识的一些优秀API开发者不会停留在像可读性这样的「第一阶层」方向上，他们将尽可能多的努力奉献给我称为「第二阶层」的API设计：**使用此API的代码如何随着时间的推移而发展**。
```

微小的需求变化可以使最优雅的代码分崩离析。

> *优秀* 的 API 会准备好这些。它们预计到你会想要移动一些代码、复制粘贴一些部分、重命名、将特殊情况提取成可复用帮助程序、将抽象展开回特定的情景、hack一下、优化瓶颈、放弃一部分然后重新开始、犯了一个错误、在起因和影响之间游走、修复bug、审查修复代码。

Reviewer: ```suggestion
*优秀*的 API 会准备好这些。它们预计到你会想要移动一些代码、复制粘贴一些部分、重命名、将特殊情况提取成可复用帮助程序、将抽象展开回特定的情景、hack 一下、优化瓶颈、放弃一部分然后重新开始、犯了一个错误、在起因和影响之间游走、修复 bug、审查修复代码。
```

> *优秀* 的APIs不仅使你走向成功，还能助你 *停留* 在那。

Reviewer: ```suggestion
*优秀*的 API 不仅使你走向成功，还能助你*停留*在那。
```

> They’re optimized for change.

Reviewer: I think line 23 is supposed to be removed? :)

> They’re optimized for change.

Reviewer: ```suggestion
```
Do you intend to keep line 23? Or did you forget to remove it? :)
